Introduce a new mechanism for applications to prepare for ractor safety (#57626)
rails/rails: activesupport/CHANGELOG.md

*   Added a new `config.active_support.unshareable_proc_action`

    Running Rails on Ractors is coming fast but not there yet. In order
    for applications to be prepared to be compatible with Ractors, we are introducing
    this configuration to control the behaviour when an application
    code passes a non shareable proc to a Rails API.

    There are 3 modes:

>     * :raise -> Rails attempt to make the proc shareable. It it can not, a loud `Ractor::IsolationError` will be raised.

Reviewer: ```suggestion
    * :raise -> Rails attempts to make the proc shareable. If it can't, a `Ractor::IsolationError` will be raised.
```

Author: Removed the changelog altogether based on Jean's feedback

>     * :warn ->  Rails attempt to make the proc shareable. If it's not, we rescue `Ractor::IsolationError` and output a warning.

Reviewer: ```suggestion
    * :warn ->  Rails attempts to make the proc shareable. If it can't, we rescue `Ractor::IsolationError` and output a warning.
```

>     * `nil` ->  Rails *do not* attempt to make the proc shareable.

Reviewer: ```suggestion
    * `nil` ->  Rails *does not* attempt to make the proc shareable.
```
